network/basic/chaincode/supplychaincc/lib/supplychaincontract.js

```diff
@@ -86,12 +94,7 @@ class SupplyChainContract extends Contract {
     async indexOrderDelivery(ctx) {
         let identity = ctx.clientIdentity;
         const enrollmentID = identity.getAttributeValue('hf.EnrollmentID');
-        // let org = identity.getMSPID();
-        // const role = "issuer";
-        // if(org === "MOPH")
-        //     role = "issuer";
-        // else
-        //     role = org;
+       
         const query = `{
             "selector": {
                 "issuer": "${enrollmentID}",
```

Comment:
border control, storage facility and hospital should have access to this function but instead of issuer, we should dynamically determine the attribute name (e.g. hospital or storage) and search for specific statuses depending on each's role